### DISCUSSION TITLE
Possibility of support for Windows XP?

### DISCUSSION THREAD

**Discussion by eddmanx** [elishacloud/dxwrapper#232] [General]
I tried it on XP but get this error:

`The procedure entry point InitializecriticalSectionEx could not be located in the dynamic link libray KERNEL32.dll.`

Any chance of a build for XP?

**Comment by elishacloud** [comment]
Not all features can be supported on Windows XP.  For example DDrawCompat won't work.  However, I can see if I can get `dd7to9` and `d3d8to9` and the Direct3D9 features available on Windows XP.

**Reply by eddmanx** [reply]
That's great. My interest is basically only in dd7to9. There's nothing like it that wraps Direct3D 2-7 to 9, well at least that I know of.
I don't think DDrawCompat is even needed for XP. It doesn't have the weird ddraw compatibility issues of modern windows.

EDIT: The standalone d3d8to9 works in XP, so it'd probably not be too difficult to make the built-in one work.

**Comment by elishacloud** [comment]
Here is a build that should work on Windows XP:  [dxwrapper.zip](https://github.com/elishacloud/dxwrapper/files/13065037/dxwrapper.zip)

There are two options that don't work: `DDrawCompat` and `DisableHighDPIScaling`.  All other features should be working.  However, I only tested `Dd7to9`.

**Reply by eddmanx** [reply]
It works, thanks. The stub seems to not support it though; throws the `Could not find DxWrapper.dll` error.

**Reply by elishacloud** [reply]
That message means that dxwrapper.dll failed to load into the program.  It could be that you are missing something needed for dxwrapper to load.

**Reply by elishacloud** [reply]
BTW: I tested this on a system with the unofficial release of Service Pack 4 installed.  This is basically Service Pack 3 plus all the released updates in a single package.

**Reply by eddmanx** [reply]
> That message means that dxwrapper.dll failed to load into the program. It could be that you are missing something needed for dxwrapper to load.

but why does dxwrapper.dll itself work when I rename it to ddraw.dll? It's just the stub ddraw.dll that's misbehaving.

I basically copied over dxwrapper.dll, dxwrapper.ini and the ddraw.dll stub, and changed dd7to9 to 1.

I'm using XP Service Pack 3.

**Reply by elishacloud** [reply]
If you rename dxwrapper.dll to ddraw.dll it still may not load.  Windows will just bypass it and load the System32 version instead.  Check if it is creating a log file.  If it is loading it will always create a log file.

**Reply by eddmanx** [reply]
It creates a log file and I see d3d9 in RTSS, so it seems to be working by itself.
[ddraw-jk.log](https://github.com/elishacloud/dxwrapper/files/13077563/ddraw-jk.log)

**Reply by eddmanx** [reply]
The stub does work on Windows 10 though.

An oddity on Windows 10 is that all the resolutions listed in the game work, except the highest, 1920x1080.
The game crashes when I try to load/start a mission.
[dxwrapper-jk.log](https://github.com/elishacloud/dxwrapper/files/13078025/dxwrapper-jk.log)

**Reply by elishacloud** [reply]
> all the resolutions listed in the game work, except the highest, 1920x1080.

Yeah, it is not just that resolution that fails.  Any resolution larger than that also fails.  I don't know the exact maximum resolution that this game can handle, but I think it is somewhere around 1440x900 or 1600x900.  Any thing larger will fail.  I believe this is an issue with the game, not dxwrapper.

> The stub does work on Windows 10 though.

The stub works fine on Windows XP for me.  The [code](https://github.com/elishacloud/dxwrapper/blob/5ddc5674b6bf96f3945272e01212b255c8ca7109/Stub/Stub.cpp#L147-L158) is quite simple.  If you get that message then the dxwrapper is not loaded.  I doubt it is a problem with the stub.  I guess it could be that dxwrapper is not loading when it is named dxwrapper.dll, but if you rename it then it can load.  There is different code that runs depending on the name.

**Comment by elishacloud** [comment]
Here is the latest build.  It works for me with both dxwrapper.dll and ddraw.dll.  The stub seems to work fine also.

[dxwrapper.zip](https://github.com/elishacloud/dxwrapper/files/13169264/dxwrapper.zip)

**Reply by eddmanx** [reply]
I must note that not all games are affected by the stub issue. Unreal Gold and Spider-Man (2001) are fine.

The stub behavior is a bit different with this version. There is no dxwrapper.dll error; it just crashes the jk.exe process at launch.

It does create a log file, though there isn't much in it:

```
1224 00:51:56.406 Starting DxWrapper v1.1.6907.22
1224 00:51:56.406 Running from: C:\GAMEFI~1\LUCASA~1\JEDIKN~1\dxwrapper.dll
1224 00:51:56.406 Reading config file: C:\GAMEFI~1\LUCASA~1\JEDIKN~1\dxwrapper.ini
```

**Reply by elishacloud** [reply]
Maybe use it without the stub?  Just rename dxwrapper.dll and dxwrapper.ini to whatever name you would have used for the stub, example ddraw.dll and ddraw.ini, respectively.  I am not sure what is the issue with the stub, since I cannot reproduce the issue.

**Comment by elishacloud** [comment]
Ok, I had an idea that might fix the stub on Windows XP.  Try this one here:  [dxwrapper.zip](https://github.com/elishacloud/dxwrapper/files/13577462/dxwrapper.zip)

**Reply by eddmanx** [reply]
No change. Could it be a VM only issue? I don't have a real XP setup to test. It works without the stub, so I guess it's not that crucial anyway.

**Comment by elishacloud** [comment]
I wonder if the issue is with dxwrapper.dll not the stub.  Can you try with this one, make sure to update the stub and dxwrapper: 
[dxwrapper.zip](https://github.com/elishacloud/dxwrapper/files/13847745/dxwrapper.zip)
